cosmwasm/packages/wasmi-runtime/src/wasm/db.rs

```diff
@@ -23,30 +23,30 @@ pub fn write_encrypted_key(
         scrambled_field_name
     );
 
-    let ad = derive_ad_for_field(&scrambled_field_name, &context)?;
+    let (ad, ad_used_gas) = derive_ad_for_field(&scrambled_field_name, &context)?;
```

Comment:
What's the reasoning for charging gas for this?

Reply:
the gas isn't charged by us. the gas is charged in cosmos because we access the storage when reading the previous key. this information is returned for tracking purposes.

Reply:
Ah, sababa!